packages/block-library/src/calendar/edit.js

```diff
@@ -8,69 +8,42 @@ import memoize from 'memize';
  * WordPress dependencies
  */
 import { Disabled } from '@wordpress/components';
-import { Component } from '@wordpress/element';
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
 import ServerSideRender from '@wordpress/server-side-render';
 
-class CalendarEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.getYearMonth = memoize( this.getYearMonth.bind( this ), {
-			maxSize: 1,
-		} );
-		this.getServerSideAttributes = memoize(
-			this.getServerSideAttributes.bind( this ),
-			{
-				maxSize: 1,
-			}
-		);
+const getYearMonth = memoize( ( date ) => {
+	if ( ! date ) {
+		return {};
 	}
+	const momentDate = moment( date );
+	return {
+		year: momentDate.year(),
+		month: momentDate.month() + 1,
+	};
+} );
 
-	getYearMonth( date ) {
-		if ( ! date ) {
-			return {};
+export default function CalendarEdit( { attributes } ) {
+	const date = useSelect( ( select ) => {
+		const coreEditorSelect = select( 'core/editor' );
+		if ( ! coreEditorSelect ) {
+			return;
 		}
-		const momentDate = moment( date );
-		return {
-			year: momentDate.year(),
-			month: momentDate.month() + 1,
-		};
-	}
-
-	getServerSideAttributes( attributes, date ) {
-		return {
-			...attributes,
-			...this.getYearMonth( date ),
-		};
-	}
+		const { getEditedPostAttribute } = coreEditorSelect;
```

Comment:
Not a blocker for merge, but since tests are broken on `master` currently, I thought I might as well point this out:

I've never seen any other place in the code where you have to check if the return value of `select( 'core/editor' )` is nullish. I know it's transferred over from the previous code, but I'm pretty sure that it was and still is completely unnecessary.

This whole section can be simplified to:

```js
const { getEditedPostAttribute } = select( 'core/editor' );
```

Reply:
@ZebulanStanphill Fixed.